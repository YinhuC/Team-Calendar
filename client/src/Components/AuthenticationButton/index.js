/* Third Party */
import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

/* Components */
import {} from './style';
import {Button} from 'reactstrap';
import {authenticate} from '../../redux/actions';

/* Functions */


class AuthenticationButton extends React.Component {
  componentDidMount() {
    fetch('/api/user_details').then((response) => {
      response.json().then((user) => {
        this.props.authenticate(user.id);
      });
    });
  }

  handleLogin = () => {
    fetch('/api/login').then((response) => {
      response.text().then((url) => {
        window.location.assign(url);
      });
    });
  }

  handleLogout = () => {
    fetch('/api/logout');
    window.location.assign('/');
  }

  render() {
    return (
      <>
        {this.props.authenticated ?
              <Button color={this.props.theme} onClick={this.handleLogout}>
                {this.props.logoutText}
              </Button>:
              <Button color={this.props.theme} onClick={this.handleLogin}>
                {this.props.loginText}
              </Button>}
      </>
    );
  }
}

AuthenticationButton.propTypes = {
  authenticated: PropTypes.bool,
  authenticate: PropTypes.func,
  logoutText: PropTypes.string,
  loginText: PropTypes.string,
  theme: PropTypes.string,
};
AuthenticationButton.defaultProps = {
  theme: 'primary',
};
function mapStateToProps(state) {
  return {
    authenticated: state.authentication.authenticated,
  };
}
const mapDispatchToProps = {
  authenticate,
};

export default connect(mapStateToProps, mapDispatchToProps)(AuthenticationButton);
