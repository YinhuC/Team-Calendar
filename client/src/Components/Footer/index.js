/* Third Party */
import React from 'react';

/* Components */
import {Logo, FooterContainer} from './style';
import {Container, Row, Col} from 'reactstrap';
import wolf from '../wolf.svg';

/* Functions */

function Footer() {
  return (
    <FooterContainer>
      <Container className="mb-5">
        <Row>
          <Col>
            <Logo src={wolf} alt="Logo" />
            <p>White Wolf</p>
          </Col>
          <Col>
            <h3>Useful Links</h3>
            <ul>
              <li>Item</li>
              <li>Item</li>
              <li>Item</li>
            </ul>
          </Col>
          <Col>
            <h3>Useful Links</h3>
            <ul>
              <li>Item</li>
              <li>Item</li>
              <li>Item</li>
            </ul>
          </Col>
        </Row>
      </Container>
      <hr />
      <Container>
        <Row>
          <Col className="mb-3">
            Â© 2020 GroupCalendar. Developed by Team White Wolf
          </Col>
        </Row>
      </Container>
    </FooterContainer>
  );
}

export default Footer;
