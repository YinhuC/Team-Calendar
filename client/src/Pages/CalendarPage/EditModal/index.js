/* Third Party */
import React from 'react';
import {
  Row, Col, Button, ModalHeader, ModalBody, ModalFooter, FormGroup, Label, Input,
} from 'reactstrap';
import {ModalStyled} from './style';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {setTimes, setDates, setTitle} from '../../../redux/actions';
import moment from 'moment';
import {Trash2} from 'react-feather';


class EditModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  editEvent = () => {
    const event = {
      summary: this.props.title,
      start: {
        dateTime: moment(`${this.props.startDate}-${this.props.startTime}`, 'YYYY-MM-DD-HH:mm').toISOString(),
      },
      end: {
        dateTime: moment(`${this.props.endDate}-${this.props.endTime}`, 'YYYY-MM-DD-HH:mm').toISOString(),
      },
    };
    fetch(`/api/calendars/${this.props.groupid}/events/${this.props.id}`, {
      method: 'PUT',
      headers: {'Content-Type': 'application/json'},
      body: JSON.stringify({event: event}),
    }).then((res, err) => {
      if (err) {
        console.log(err);
      }
      this.props.refresh();
    });
    this.closeModal();
  }

  closeModal = () =>{
    this.props.toggle();
    this.props.setTitle(
        '',
    );
  }

  render() {
    return (
      <ModalStyled size="lg" isOpen={this.props.isOpen} toggle={this.props.toggle}>
        <ModalHeader toggle={this.props.toggle} >
          <span style={{display: 'flex', flexWrap: 'nowrap'}}>
            <span className="mr-1" >
            Edit Calendar Event
            </span>
            <Button color="clear" style={{padding: '0', display: 'flex', alignItems: 'center'}}>
              <Trash2 size={25}/>
            </Button>
          </span>
        </ModalHeader>
        <ModalBody>
          <Row>
            <Col className="col-12">
              <FormGroup>
                <Label>Name of Event</Label>
                <Input
                  type="text"
                  name="event"
                  placeholder="Name of event"
                  onChange={(event) => this.props.setTitle(event.target.value, this.props.title)}
                  value={this.props.title}
                />
              </FormGroup>
            </Col>
            <Col>
              <FormGroup>
                <Label>Start Date</Label>
                <Input
                  type="date"
                  name="startDate"
                  onChange={(event) => this.props.setDates(event.target.value, this.props.endDate)}
                  value={this.props.startDate}
                />
              </FormGroup>
              <FormGroup>
                <Label>End Date</Label>
                <Input
                  type="date"
                  name="endDate"
                  onChange={(event) => this.props.setDates(this.props.startDate, event.target.value)}
                  value={this.props.endDate}
                />
              </FormGroup>
            </Col>
            <Col>
              <FormGroup>
                <Label >Start Time</Label>
                <Input
                  type="time"
                  name="startTime"
                  onChange={(event) => this.props.setTimes(event.target.value, this.props.endTime)}
                  value={this.props.startTime}
                />
              </FormGroup>
              <FormGroup>
                <Label>End Time</Label>
                <Input
                  type="time"
                  name="endTime"
                  onChange={(event) => this.props.setTimes(this.props.startTime, event.target.value)}
                  value={this.props.endTime}
                />
              </FormGroup>
            </Col>
          </Row>
        </ModalBody>

        <ModalFooter>
          <Button color="primary" onClick={this.editEvent}>Save</Button>
          <Button color="secondary" onClick={this.props.toggle}>Cancel</Button>
        </ModalFooter>
      </ModalStyled>);
  }
}
EditModal.propTypes = {
  toggle: PropTypes.func,
  refresh: PropTypes.func,
  isOpen: PropTypes.bool,
  groupid: PropTypes.string,
  startDate: PropTypes.string,
  endDate: PropTypes.string,
  startTime: PropTypes.string,
  endTime: PropTypes.string,
  setTimes: PropTypes.func,
  setDates: PropTypes.func,
  title: PropTypes.string,
  setTitle: PropTypes.func,
  id: PropTypes.string,
};
function mapStateToProps(state) {
  return {
    startDate: state.eventModal.startDate,
    endDate: state.eventModal.endDate,
    startTime: state.eventModal.startTime,
    endTime: state.eventModal.endTime,
    title: state.eventModal.title,
    id: state.eventModal.id,
  };
}
const mapDispatchToProps = {
  setTimes, setDates, setTitle,
};
export default connect(mapStateToProps, mapDispatchToProps)(EditModal);
