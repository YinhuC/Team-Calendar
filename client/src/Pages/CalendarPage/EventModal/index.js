/* Third Party */
import React from 'react';
import {
  Row, Col, Button, ModalHeader, ModalBody, ModalFooter, FormGroup, Label, Input,
} from 'reactstrap';
import {ModalStyled} from './style';
import moment from 'moment';
class EventModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      startDate: '',
      startTime: '',
      endDate: '',
      endTime: '',
      event: '',
    };
  }
  changeInput = (event) => {
    this.setState({
      [event.target.name]: event.target.value,
    });
  }
  render() {
    return (
      <ModalStyled size="lg" isOpen={this.props.isOpen} toggle={this.props.toggle}>
        <ModalHeader>Add New Calendar Event</ModalHeader>
        <ModalBody>
          <Row>
            <Col className="col-12">
              <FormGroup>
                <Label>Name of Event</Label>
                <Input
                  type="text"
                  name="event"
                  placeholder="Name of event"
                  onChange={this.changeInput}
                  value={this.state.event}
                />
              </FormGroup>
            </Col>
            <Col>
              <FormGroup>
                <Label>Start Date</Label>
                <Input
                  type="date"
                  name="startDate"
                  onChange={this.changeInput}
                  value={moment(this.state.startDate).format('YYYY-MM-DD')}
                />
              </FormGroup>
              <FormGroup>
                <Label>End Date</Label>
                <Input
                  type="date"
                  name="endDate"
                  onChange={this.changeInput}
                  value={moment(this.state.endDate).format('YYYY-MM-DD')}
                />
              </FormGroup>
            </Col>
            <Col>
              <FormGroup>
                <Label >Start Time</Label>
                <Input
                  type="time"
                  name="startTime"
                  onChange={this.changeInput}
                  value={moment(this.state.startTime).format('HH:mm')}
                />
              </FormGroup>
              <FormGroup>
                <Label>End Time</Label>
                <Input
                  type="time"
                  name="endTime"
                  onChange={this.changeInput}
                  value={moment(this.state.endTime).format('HH:mm')}
                />
              </FormGroup>
            </Col>
          </Row>
        </ModalBody>

        <ModalFooter>
          <Button color="primary" onClick={this.props.toggle}>Add</Button>
          <Button color="secondary" onClick={this.props.toggle}>Cancel</Button>
        </ModalFooter>
      </ModalStyled>);
  }
}

export default EventModal;
