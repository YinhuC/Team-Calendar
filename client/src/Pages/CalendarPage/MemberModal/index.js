/* Third Party */
import React from 'react';
import {
  Row, Col, Button, ModalHeader, ModalBody, ModalFooter, FormGroup, Label, Input,
} from 'reactstrap';
import {ModalStyled} from './style';
import PropTypes from 'prop-types';

class MemberModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
    };
  }


  changeInput = (event) => {
    this.setState({
      [event.target.name]: event.target.value,
    });
  }

  addMember = () => {
    const {groupid} = this.props;
    fetch('/api/groups/'+groupid, {
      method: 'PUT',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        email: this.state.email,
      }),
    }).then((res, err) => {
      if (err) {
        console.log(err);
      }
      if (res.status === 404) {
        alert('User not found.');
      }
      this.props.refresh();
    });
    this.props.toggle();
    this.closeModal();
  }

  closeModal = () => {
    this.setState({
      email: '',
    });
    this.props.toggle();
  }

  render() {
    return (
      <ModalStyled size="lg" isOpen={this.props.isOpen} toggle={this.props.toggle}>
        <ModalHeader toggle={this.closeModal}>Add New Member</ModalHeader>
        <ModalBody>
          <Row>
            <Col className="col-12">
              <FormGroup>
                <Label>New Member Email</Label>
                <Input
                  type="email"
                  name="email"
                  placeholder="Enter email address here"
                  onChange={this.changeInput}
                  value={this.state.email}
                />
              </FormGroup>
            </Col>
          </Row>
        </ModalBody>
        <ModalFooter>
          <Button color="primary" onClick={this.addMember}>Add</Button>
          <Button color="secondary" onClick={this.closeModal}>Cancel</Button>
        </ModalFooter>
      </ModalStyled>);
  }
}
MemberModal.propTypes = {
  groupid: PropTypes.string,
  toggle: PropTypes.func,
  isOpen: PropTypes.bool,
  refresh: PropTypes.func,
};
export default MemberModal;
