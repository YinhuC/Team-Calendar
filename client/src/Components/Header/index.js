/* Third Party */
import React from 'react';
import {Row, Col} from 'reactstrap';
import wolf from '../wolf.svg';
import wolfLight from '../wolf_light.svg';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

/* Components */
import {Center, Logo} from './style';
import AuthenticationButton from '../AuthenticationButton';

/* Functions */

class Header extends React.Component {
  render() {
    return (
      <Center style={!this.props.authenticated?{backgroundColor: '#007bff'}:{}}
        className="d-flex justify-content-around">
        <Row className="mr-5">
          <Col className="align-middle">
            <Logo src={this.props.authenticated? wolf:wolfLight} alt="Logo" />
          </Col>
        </Row>
        <Row><Col></Col></Row>
        <Row className="ml-5">
          <Col className="align-middle">
            <AuthenticationButton theme = {this.props.authenticated ?'primary':'light'}
              loginText="Sign In" logoutText="Logout"/>
          </Col>
        </Row>
      </Center>
    );
  }
}

Header.propTypes = {
  authenticated: PropTypes.bool,
};
function mapStateToProps(state) {
  return {
    authenticated: state.authentication.authenticated,
  };
}

export default connect(mapStateToProps)(Header);
