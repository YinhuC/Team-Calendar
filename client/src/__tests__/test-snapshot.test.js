/* Third Party */
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

/* Components */
import Header from '../Components/Header';
import Footer from '../Components/Footer';
import AuthenticationButton from '../Components/AuthenticationButton';
import CalendarPage from '../Pages/CalendarPage';
import {MemoryRouter} from 'react-router';
import EditModal from '../Pages/CalendarPage/EditModal';
import EventModal from '../Pages/CalendarPage/EventModal';
import MemberModal from '../Pages/CalendarPage/MemberModal';
import ErrorPage from '../Pages/ErrorPage';
import LandingPage from '../Pages/LandingPage';
import DetailSection from '../Pages/MarketingPage/DetailSection';
import FeatureSection from '../Pages/MarketingPage/FeatureSection';
import HeaderSection from '../Pages/MarketingPage/HeaderSection';

const mockStore = configureStore([]);

it('testing the AuthButton no Auth', () => {
  const store = mockStore({
    authentication: {authenticated: false},
  });
  const tree = renderer.create(<Provider store={store}><AuthenticationButton/></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('testing the AuthButton Auth', () => {
  const store = mockStore({
    authentication: {authenticated: true},
  });
  const tree = renderer.create(<Provider store={store}><AuthenticationButton/></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});


it('testing the Header no auth', () => {
  const store = mockStore({
    authentication: {authenticated: false},
  });
  const tree = renderer.create(<Provider store={store}><Header/></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('testing the Header auth', () => {
  const store = mockStore({
    authentication: {authenticated: true},
  });
  const tree = renderer.create(<Provider store={store}><Header/></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('testing the footer', () => {
  const tree = renderer.create(<Footer/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('testing the CalendarPage', ()=>{
  const store = mockStore({
    eventModal: {startDate: ''},
  });
  const tree = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <CalendarPage match={{params: {googleId: ''}}}/>
        </Provider>
      </MemoryRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('testing the EditModalPage', ()=>{
  const store = mockStore({
    eventModal: {startDate: ''},
  });
  const tree = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <EditModal/>
        </Provider>
      </MemoryRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
it('testing the EventModalPage', ()=>{
  const store = mockStore({
    eventModal: {startDate: ''},
  });
  const tree = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <EventModal/>
        </Provider>
      </MemoryRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
it('testing the MemberModalPage', ()=>{
  const tree = renderer.create(
      <MemoryRouter>
        <MemberModal/>
      </MemoryRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
it('testing the ErrorPage', ()=>{
  const tree = renderer.create(
      <MemoryRouter>
        <ErrorPage/>
      </MemoryRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
it('testing the LandingPage', ()=>{
  const tree = renderer.create(
      <MemoryRouter>
        <LandingPage/>
      </MemoryRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
// Marketing Page is skipped as it only contains the sub sections.
it('testing the HeaderSection', () => {
  const store = mockStore({
    authentication: {authenticated: false},
  });
  const tree = renderer.create(<Provider store={store}><HeaderSection/></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
it('testing the DetailSection', () => {
  const tree = renderer.create(<DetailSection/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('testing the FeatureSection', () => {
  const tree = renderer.create(<FeatureSection/>).toJSON();
  expect(tree).toMatchSnapshot();
});
