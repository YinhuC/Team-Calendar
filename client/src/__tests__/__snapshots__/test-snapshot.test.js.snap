// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the AuthButton Auth 1`] = `
<button
  aria-label={null}
  className="btn btn-primary"
  onClick={[Function]}
  type="button"
/>
`;

exports[`testing the AuthButton no Auth 1`] = `
<button
  aria-label={null}
  className="btn btn-primary"
  onClick={[Function]}
  type="button"
/>
`;

exports[`testing the CalendarPage 1`] = `
<div
  className="sc-AxgMl jLVCgA flex-lg-row flex-column"
>
  <div
    className="sc-AxheI gocOSI col-lg-2 col-12"
  >
    <div
      className="row"
    >
      <div
        className="col-12 d-flex justify-content-center my-3 col"
      >
        <a
          href="/landing"
          onClick={[Function]}
        >
          <button
            aria-label={null}
            className="btn btn-primary"
            onClick={[Function]}
          >
            &lt; Dashboard
          </button>
        </a>
      </div>
      <div
        className="col-12 col"
      >
        <div
          className="sc-fzplWN Aphti"
        >
          Calendars
        </div>
        <div
          className="sc-AxmLO JCFsi"
        >
          <ul
            className="sc-fzoLsD dfXHGQ list-group"
          />
        </div>
      </div>
      <div
        className="col-12 mb-1 d-flex justify-content-between col"
      >
        <div
          className="sc-fzplWN Aphti"
        >
          Members
        </div>
        <button
          aria-label={null}
          className="btn btn-primary"
          onClick={[Function]}
          type="button"
        >
          <svg
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="5"
              y2="19"
            />
            <line
              x1="5"
              x2="19"
              y1="12"
              y2="12"
            />
          </svg>
        </button>
      </div>
      <div
        className="col-12 col"
      >
        <div
          className="sc-fzozJi dsYdeD"
        >
          <ul
            className="sc-fzoLsD dfXHGQ list-group"
          >
            <li
              className="sc-fzpans itRoqh list-group-item"
              style={
                Object {
                  "backgroundColor": "#091697",
                  "color": "#FFFFFF",
                }
              }
            >
              
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-fzokOt fECDVp col-lg-10 col-12"
  >
    <div
      className="sc-fznZeY ZJQKm"
    >
      <div
        className="row"
      >
        <div
          className="col-12 d-flex space-between align-items-center col"
        >
          <div
            className="sc-Axmtr gLsYaZ"
          >
            
          </div>
          <button
            aria-label={null}
            className="sc-fzqBZW latBVY btn btn-primary"
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="none"
              height={24}
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width={24}
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="12"
                x2="12"
                y1="5"
                y2="19"
              />
              <line
                x1="5"
                x2="19"
                y1="12"
                y2="12"
              />
            </svg>
            <span
              className="d-none d-xl-block"
            >
              Add Event
            </span>
          </button>
        </div>
        <div
          className="col-12 col"
        >
          <div
            className="sc-fznKkj jhGIKY"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`testing the DetailSection 1`] = `
Array [
  <div
    className="sc-fzqPZZ kVVlLP"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12 col-lg-8 mb-4 mt-xl-5 order-1 order-lg-2 col"
        >
          <img
            className="sc-fzoYkl dwIITi"
            src="create_group.gif"
          />
        </div>
        <div
          className="col-12 col-lg-4 mt-xl-5 order-2 order-lg-1 col"
        >
          <div
            className="sc-fzoxKX iuQDVH"
          >
            Create Groups
          </div>
          <div
            className="sc-fzoKki Imobz"
          >
            Sign in with your Google Account and create your group with a single click!
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-fzqBkg JWBHk"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12 col-lg-4 order-2 col"
        >
          <div
            className="sc-fzoxKX iuQDVH"
          >
            Add Members
          </div>
          <div
            className="sc-fzoKki Imobz"
          >
            Type in your members Gmail Address and instanly add them to the group!
          </div>
        </div>
        <div
          className="col-12 col-lg-8 mb-4 col"
        >
          <img
            className="sc-fzoYkl dwIITi"
            src="add_member.gif"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-fzqPZZ kVVlLP"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12 col-lg-8 mb-4 order-1 order-lg-2 col"
        >
          <img
            className="sc-fzoYkl dwIITi"
            src="select_calendar.gif"
          />
        </div>
        <div
          className="col-12 col-lg-4 mt-xl-5 order-2 order-lg-1 col"
        >
          <div
            className="sc-fzoxKX iuQDVH"
          >
            Choose What To Share
          </div>
          <div
            className="sc-fzoKki Imobz"
          >
            Select which if your Google Calendars you want to share with your group!
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-fzqBkg JWBHk"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12 col-lg-8 mb-4 col"
        >
          <img
            className="sc-fzoYkl dwIITi"
            src="new_event.gif"
          />
        </div>
        <div
          className="col-12 col-lg-4 mt-xl-5 col"
        >
          <div
            className="sc-fzoxKX iuQDVH"
          >
            Schedule Meetings
          </div>
          <div
            className="sc-fzoKki Imobz"
          >
            Select on the calendar to invite everyone to your meeting!
          </div>
        </div>
      </div>
    </div>
  </div>,
  <svg
    className="sc-fzolEj fbHTzo"
  >
    wave_bottom.svg
  </svg>,
]
`;

exports[`testing the EditModalPage 1`] = `null`;

exports[`testing the ErrorPage 1`] = `
<div
  className="sc-fznxsB hQUfhz"
>
   Error 404
</div>
`;

exports[`testing the EventModalPage 1`] = `null`;

exports[`testing the FeatureSection 1`] = `
<div
  className="sc-fzpkJw bAYsQo"
>
  <div
    className="container"
  >
    <div
      className="d-flex justify-content-center row"
    >
      <div
        className="col-lg-3 col-12 col"
      >
        <div
          className="sc-fzomME fWoUld"
        >
          <img
            className="sc-fzoant iRMKOZ"
            src="calendar.png"
          />
          <div
            className="sc-fznzOf flBNCK"
          >
            Complete
          </div>
          <div
            className="sc-fznMnq iLxXRK"
          >
            See all of your group members' schedules in one place
          </div>
        </div>
      </div>
      <div
        className="col-lg-3 col-12 col"
      >
        <div
          className="sc-fzomME fWoUld"
        >
          <img
            className="sc-fzoant iRMKOZ"
            src="simple.png"
          />
          <div
            className="sc-fznzOf flBNCK"
          >
            Simple
          </div>
          <div
            className="sc-fznMnq iLxXRK"
          >
            Organise a meeting that works for everyone in one click
          </div>
        </div>
      </div>
      <div
        className="col-lg-3 col-12 col"
      >
        <div
          className="sc-fzomME fWoUld"
        >
          <img
            className="sc-fzoant iRMKOZ"
            src="google.svg"
          />
          <div
            className="sc-fznzOf flBNCK"
          >
            Integrated
          </div>
          <div
            className="sc-fznMnq iLxXRK"
          >
            All your group events are synced to your Google Account
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`testing the Header auth 1`] = `
<div
  className="sc-AxjAm hpCAWI d-flex justify-content-around"
  style={Object {}}
>
  <div
    className="mr-5 row"
  >
    <div
      className="align-middle col"
    >
      <img
        alt="Logo"
        className="sc-AxirZ igrhLV"
        src="wolf_dark.svg"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    />
  </div>
  <div
    className="ml-5 row"
  >
    <div
      className="align-middle col"
    >
      <button
        aria-label={null}
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Logout
      </button>
    </div>
  </div>
</div>
`;

exports[`testing the Header no auth 1`] = `
<div
  className="sc-AxjAm hpCAWI d-flex justify-content-around"
  style={
    Object {
      "backgroundColor": "#007bff",
      "boxShadow": "none",
      "position": "fixed",
      "top": "0",
    }
  }
>
  <div
    className="mr-5 row"
  >
    <div
      className="align-middle col"
    >
      <img
        alt="Logo"
        className="sc-AxirZ igrhLV"
        src="wolf_light.svg"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    />
  </div>
  <div
    className="ml-5 row"
  >
    <div
      className="align-middle col"
    >
      <button
        aria-label={null}
        className="btn btn-light"
        onClick={[Function]}
        type="button"
      >
        Sign In
      </button>
    </div>
  </div>
</div>
`;

exports[`testing the HeaderSection 1`] = `
Array [
  <div
    className="sc-fzqMAW qWTko"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="sc-fzoydu xlRdo"
          >
            <div
              className="sc-fzoJMP eKUMdk"
            >
              The Calendar With Everyone, For Everyone
            </div>
            <div
              className="sc-fzoYHE hnStfN d-none d-md-flex"
            >
              Gain access to the schedule of your group and create events which are integrated with Google Calendars.
            </div>
            <button
              aria-label={null}
              className="btn btn-light"
              onClick={[Function]}
              type="button"
            >
              Sign Up & Start  &gt;
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <svg
    className="sc-fzqAbL exggMT"
  >
    wave_top.svg
  </svg>,
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="d-flex justify-content-center  col"
      >
        <img
          className="sc-fzpkqZ cetbdm"
          src="calendar.png"
        />
      </div>
    </div>
  </div>,
]
`;

exports[`testing the LandingPage 1`] = `
<div
  className="sc-fznWqX fdDzKF"
>
  <div
    className="sc-fzoiQi oHwyo"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="sc-fzqARJ fANjuf d-flex text-center text-lg-left"
        >
          Dashboard
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12 col-xl-4 col-lg-6 col"
      >
        <div
          className="sc-fznJRM hxehQo card"
        >
          <button
            aria-label={null}
            className="btn btn-outline-primary"
            onClick={[Function]}
            style={
              Object {
                "borderRadius": 10,
                "flex": 1,
              }
            }
            type="button"
          >
            Create Group
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`testing the MemberModalPage 1`] = `null`;

exports[`testing the footer 1`] = `
<div
  className="sc-AxiKw fRmbi"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-12 col-md-6 d-flex justify-content-center col"
      >
        <img
          alt="Logo"
          className="sc-AxhUy cpmCoU"
          src="white_text.png"
        />
      </div>
      <div
        className="col-12 col-md-6 pl-5 d-flex justify-content-center align-items-center col"
      >
        <h3>
          This web application was designed for a group project in Semester One 2020 at the University of Auckland.
        </h3>
      </div>
    </div>
  </div>
  <hr
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  />
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="mb-3 col"
      >
        © 2020 GroupCalendar. Developed by Team White Wolf
      </div>
    </div>
  </div>
</div>
`;
